### PR TITLE
Assignment controller

### PR DESCRIPTION
**Pull request by lena-aiu**
@jrmcgarvey Hi John, please review the code. 
Made changes in access control for assignments, changed show view for clients with a new assignment or new order base on the role.  Also, I made changes in the request test for categories post with invalid data. In this version, we don't have the latest changes in Navigation. All changes with navbar in branch - navigationNew.
Thank you

### CODE REVIEW

File: app/controllers/services_controller.rb [hunk 1]
```diff
@@ -13,57 +13,88 @@ class ServicesController < ApplicationController
...
+        marker.lng service.longitude
+        marker.infowindow "<a href='https://www.google.com/maps/dir/Current+Location/#{service.address}' target='_blank'>#{service.name}</a>"
+      end
   end
 
   # GET /serivices/new
   def new
-    @service = Service.new
+    # @service = Service.new
+    if  nil? || volunteer?
```
Comment (jrmcgarvey) [inline app/controllers/services_controller.rb:37]: problem here (regression)

File: app/controllers/services_controller.rb [hunk 2]
```diff
@@ -13,57 +13,88 @@ class ServicesController < ApplicationController
...
+      flash.notice = "You are not authorized for that operation."
+      redirect_to home_index_path
+    end
+
   end
 
   # GET /services/1/edit
   def edit
-
+    if  nil? || volunteer?
```
Comment (jrmcgarvey) [inline app/controllers/services_controller.rb:51]: problem here (regression)

File: app/controllers/services_controller.rb [hunk 3]
```diff
@@ -13,57 +13,88 @@ class ServicesController < ApplicationController
...
+    else
+      flash.notice = "You are not authorized for that operation."
+      redirect_to home_index_path
+    end
   end
 
   # POST /services
   # POST /services.json
   def create
+    if  nil? || volunteer?
```
Comment (jrmcgarvey) [inline app/controllers/services_controller.rb:65]: problem here (regression of fix for nil? problem)

File: app/controllers/services_controller.rb [hunk 4]
```diff
@@ -13,57 +13,88 @@ class ServicesController < ApplicationController
...
       flash.now.alert = @service.errors.full_messages.to_sentence
       render :edit
     end
+
   end
 
   # PATCH/PUT /customers/1
   # PATCH/PUT /customers/1.json
   def update
+    if  nil? || volunteer?
```
Comment (jrmcgarvey) [inline app/controllers/services_controller.rb:88]: same here

File: app/controllers/services_controller.rb [hunk 5]
```diff
@@ -75,6 +106,15 @@ def update
   # DELETE /customers/1
   # DELETE /customers/1.json
   def destroy
+    if  nil? || volunteer?
```
Comment (jrmcgarvey) [inline app/controllers/services_controller.rb:109]: same here

**Review by jrmcgarvey — CHANGES_REQUESTED** [review]
We have regression of fixes from PR 89, with the if nil? error.  This may get fixed if the changes from the master are merged and you push again.

File: app/controllers/services_controller.rb (hunk 1 above)
Comment (lena-aiu) [reply app/controllers/services_controller.rb:37]: It got fixed after merging the changes from the master

File: app/controllers/services_controller.rb (hunk 3 above)
Comment (lena-aiu) [reply app/controllers/services_controller.rb:65]: It got fixed after merging the changes from the master

File: app/controllers/services_controller.rb (hunk 4 above)
Comment (lena-aiu) [reply app/controllers/services_controller.rb:88]: It got fixed after merging the changes from the master

File: app/controllers/services_controller.rb (hunk 5 above)
Comment (lena-aiu) [reply app/controllers/services_controller.rb:109]: It got fixed after merging the changes from the master

File: app/controllers/services_controller.rb (hunk 2 above)
Comment (lena-aiu) [reply app/controllers/services_controller.rb:51]: It got fixed after merging the changes from the master

**Comment by lena-aiu** [comment]
@jrmcgarvey Hi John, all got fixed after merging the changes from the master. Also, I made small changes in navigation, using flex property to place English/Spanish on the same line as placed tabs for Home, About us, etc. And the responsiveness of the navbar working better, because not leaving the email address of the user outside the web page, which was resized.

**Review by jrmcgarvey — APPROVED** [review]
looks ok

**Pull request merged**